src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.NonAndroid.cs

```diff
@@ -72,6 +72,11 @@ private static TimeZoneInfoResult TryGetTimeZoneFromLocalMachineCore(string id,
             value = null;
             e = null;
 
+            if (Invariant)
+            {
```

Comment:
should assert here not having Invariant is true?

Reply:
You are saying that none of the `TryGetTimeZoneFromLocalMachine` or `TryGetTimeZoneFromLocalMachineCore` methods should be reachable when Invariant is true. You are right. I will replace it with assert.

Reply:
Testing with `if (Invariant) ThrowHelper.ThrowUnreachableException();`